core/services/ocr2/plugins/ocr2keeper/custom_telemetry.go

```diff
@@ -0,0 +1,152 @@
+package ocr2keeper
+
+import (
+	"context"
+	"encoding/hex"
+	"time"
+
+	"github.com/smartcontractkit/libocr/commontypes"
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	evm21 "github.com/smartcontractkit/chainlink/v2/core/services/ocr2/plugins/ocr2keeper/evm21"
+	"github.com/smartcontractkit/chainlink/v2/core/services/synchronization/telem"
+	"github.com/smartcontractkit/chainlink/v2/core/static"
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+	ocr2keepers "github.com/smartcontractkit/ocr2keepers/pkg/v3/types"
+)
+
+type AutomationCustomTelemetryService struct {
+	utils.StartStopOnce
+	monitoringEndpoint    commontypes.MonitoringEndpoint
+	blockSubscriber       *evm21.BlockSubscriber
+	blockSubChanID        int
+	threadCtrl            utils.ThreadControl
+	lggr                  logger.Logger
+	configDigest          [32]byte
+	contractConfigTracker types.ContractConfigTracker
+}
+
+// NewAutomationCustomTelemetryService creates a telemetry service for new blocks and node version
+func NewAutomationCustomTelemetryService(me commontypes.MonitoringEndpoint,
+	lggr logger.Logger, blocksub *evm21.BlockSubscriber, configTracker types.ContractConfigTracker) (*AutomationCustomTelemetryService, error) {
+	return &AutomationCustomTelemetryService{
+		monitoringEndpoint:    me,
+		threadCtrl:            utils.NewThreadControl(),
+		lggr:                  lggr.Named("AutomationCustomTelem"),
+		contractConfigTracker: configTracker,
+		blockSubscriber:       blocksub,
+	}, nil
+}
+
+// Start starts Custom Telemetry Service, sends 1 NodeVersion message to endpoint at start and sends new BlockNumber messages
+func (e *AutomationCustomTelemetryService) Start(ctx context.Context) error {
+	return e.StartOnce("AutomationCustomTelemetryService", func() error {
+		e.lggr.Infof("Starting: Custom Telemetry Service")
+		_, configDetails, err := e.contractConfigTracker.LatestConfigDetails(ctx)
+		if err != nil {
+			e.lggr.Errorf("Error occurred while getting newestConfigDetails for initialization %s", err)
+		} else {
+			e.configDigest = configDetails
+			e.sendNodeVersionMsg()
+		}
+		e.threadCtrl.Go(func(ctx context.Context) {
+			ticker := time.NewTicker(1 * time.Minute)
+			defer ticker.Stop()
+			for {
+				select {
+				case <-ticker.C:
+					_, newConfigDigest, err := e.contractConfigTracker.LatestConfigDetails(ctx)
+					if err != nil {
+						e.lggr.Errorf("Error occurred while getting newestConfigDetails in configDigest loop %s", err)
+					}
+					if newConfigDigest != e.configDigest {
+						e.configDigest = newConfigDigest
+						e.sendNodeVersionMsg()
+					}
+				case <-ctx.Done():
+					return
+				}
+			}
+		})
+
+		chanID, blockSubscriberChan, blockSubErr := e.blockSubscriber.Subscribe()
+		if blockSubErr != nil {
```

Comment:
i wonder if this should be a case for just returning error here and not let the whole service start instead of silently not starting block number loop. cc @amirylm for thoughts on how we handle such issues in other services